### ISSUE TITLE
[Request] Scripting API - Allow tools that are grouped in a drop-down

### ISSUE THREAD

**Issue by BlaXun** [mapeditor/tiled#3241]
I am creating a custom tool that places a "warp object" (custom object with custom properties). I do have different types of warps, but having all those different tools in the toolbar is kinda cramped. I think it would be nice if a tool could function as a dropdown... or there was a ToolSection-Object which would then display collapsed and the user could press the tool to get the actual selection of nested tools to choose from.

**Comment by bjorn** [comment]
Right, this is a feature that could be nice to have also for some of the built-in tools.
